### PR TITLE
Line merge collinear line

### PR DESCRIPTION
**Pull request by mwojcik-sagiton**
BACKGROUND:
I found `MergeCollinearLine(Line line)` method useful and missing in `Line` class 

DESCRIPTION:
Method `MergeCollinearLine(Line line)` takes all 4 vertices of two lines, sort them and creates line which contains all of them. Finally it keeps direction of original line.

TESTING:
Since there are many similarities to `TryGetOverlap(....)` method I reused unit test cases.  
  
FUTURE WORK:
None

REQUIRED:
- [x] All changes are up to date in `CHANGELOG.md`.

COMMENTS:
None

---
This change is [Reviewable](https://reviewable.io/reviews/hypar-io/elements/810)

### CODE REVIEW

File: Elements/src/Geometry/Line.cs [hunk 1]
```diff
@@ -1045,6 +1045,30 @@ public bool TryGetOverlap(Line line, out Line overlap)
             return true;
         }
 
+        /// <summary>
+        /// Creates new line with verticies of current and joined line
```
Comment (ikeough) [inline Elements/src/Geometry/Line.cs:1049]: "...with _vertices_ of current and joined..."

File: Elements/src/Geometry/Line.cs [hunk 2]
```diff
@@ -1045,6 +1045,30 @@ public bool TryGetOverlap(Line line, out Line overlap)
...
+        /// </summary>
+        /// <param name="line">Collinear line</param>
+        /// <returns>New line containing verticies of all merged lines</returns>
```
Comment (ikeough) [inline Elements/src/Geometry/Line.cs:1052]: "...containing _vertices_ of all..."

File: Elements/src/Geometry/Line.cs [hunk 3]
```diff
@@ -1045,6 +1045,30 @@ public bool TryGetOverlap(Line line, out Line overlap)
...
+        /// <exception cref="ArgumentException">Throws exception when lines are not collinear</exception>
+        public Line MergeCollinearLine(Line line)
```
Comment (ikeough) [inline Elements/src/Geometry/Line.cs:1054]: Can we call this `MergedWithCollinear`. In Elements we use the past tense for methods that result in a new object.

File: Elements/src/Geometry/Line.cs (hunk 1 above)
Comment (mwojcik-sagiton) [reply Elements/src/Geometry/Line.cs]: Fixed. Sorry for this silly spelling, I added spell checker tool to my Visual Studio

File: Elements/src/Geometry/Line.cs (hunk 3 above)
Comment (mwojcik-sagiton) [reply Elements/src/Geometry/Line.cs]: done [here](https://github.com/hypar-io/Elements/pull/810/commits/131f95956f60df5300cff98d5738e98a32623a57)

**Review by wynged — CHANGES_REQUESTED** [review]
> Reviewed 1 of 3 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/hypar-io/elements/810)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @ikeough and @mwojcik-sagiton)

___
*[`Elements/src/Geometry/Line.cs` line 1056 at r2](https://reviewable.io/reviews/hypar-io/elements/810#-N2nn96u9ynBQf4ezgtQ:-N2nn96u9ynBQf4ezgtR:b7nzqzl) ([raw file](https://github.com/hypar-io/elements/blob/131f95956f60df5300cff98d5738e98a32623a57/Elements/src/Geometry/Line.cs#L1056)):*
> ```Smalltalk
>         public Line MergedCollinearLine(Line line)
>         {
>             if (!IsCollinear(line))
> ```

curley braces around if expression.

**Review by mwojcik-sagiton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hypar-io/elements/810)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @ikeough, @mwojcik-sagiton, and @wynged)

___
*[`Elements/src/Geometry/Line.cs` line 1056 at r2](https://reviewable.io/reviews/hypar-io/elements/810#-N2nn96u9ynBQf4ezgtQ:-N2oe1Aw0bkfPEaebzsT:b-896fix) ([raw file](https://github.com/hypar-io/elements/blob/131f95956f60df5300cff98d5738e98a32623a57/Elements/src/Geometry/Line.cs#L1056)):*
<details><summary><i>Previously, wynged (Eric Wassail) wrote…</i></summary><blockquote>

curley braces around if expression.
</blockquote></details>

Done.

**Review by wynged — CHANGES_REQUESTED** [review]
> Reviewed 2 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/hypar-io/elements/810)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @ikeough and @mwojcik-sagiton)

___
*[`Elements/test/LineTests.cs` line 586 at r5](https://reviewable.io/reviews/hypar-io/elements/810#-N2qdGLr678sg_mvACNT:-N2qdGLr678sg_mvACNU:b-iubphs) ([raw file](https://github.com/hypar-io/elements/blob/beb57a40a7506281a50356a23bd1c007b2a9bccf/Elements/test/LineTests.cs#L586)):*
> ```Smalltalk
>         }
> 
>         [MemberData(nameof(ProjectedData))]
> ```

doesn't this need to be a Theory for the MemberData to do anything?

**Review by wynged — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/hypar-io/elements/810)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @ikeough)

**Pull request merged**